src/main/java/hu/dungeonhunter/tools/Fight.java

```diff
@@ -36,72 +46,74 @@ public Fight(int numOfMonsters) {  //for test
+        if (monster.getType().equals(CharacterTypes.GOBLIN_KING)) {
+            System.out.println("You can't run away from the " + monster.getType().charType + " !");
+            runningAwayFromGoblinKingTest = true; //to test
+        } else {
+            System.out.println("The monster hits you a last time before you can run away: ");
+            champion.setHp(champion.getHp() - monster.getMonsterDamage());
+            System.out.println("Champion have now " + champion.getHp() + " hit points");
+            champion.enemyVictory();
+            if (!champion.isDefeat()) {
+                monsterCounter--;
+                System.out.println("Your escape was successful! You can go further in the cave.");
+                monsterIncomingOrWin();
+            }
         }
-
     }
 
     private void monsterIncomingOrWin() {
         if (monsterCounter != 0) {
-            System.out.println("Still left in the cave " + monsterCounter + " !");
+            System.out.println(monsterCounter + " monsters are in the Dungeon!");
+            System.out.print("Something is coming! ");
+            randomEnemy = Dice.rollDice(monsterCounter, 1);
             monsterCaller();
         } else
             System.out.println("The Dungeon is clear! You killed " + killedMonsterCounter +
                     " monster (not counting the many mothers and children), you win!");
     }
 
-    public Monsters monsterCaller() {
-        if (monsterCounter != 0) {
-            System.out.println("A monster steps out from darkness!");
-            monster = new Monsters();
+    public MonstersInterface monsterCaller() {
+        if (randomEnemy == 1) {
+            monster = monsterFactory.getMonster(CharacterTypes.GOBLIN_KING);
+            goblinKingDamage();
+        } else {
+            System.out.println("A goblin steps out from the darkness!");
+            monster = monsterFactory.getMonster(CharacterTypes.GOBLIN);
         }
         return monster;
     }
 
     public void monsterAttack() {
         System.out.print("Monster attack: ");
-        champion.setHp(champion.getHp() - monster.monsterDamage());
+        champion.setHp(champion.getHp() - monster.getMonsterDamage());
         System.out.println("Champion have now " + champion.getHp() + " hit points");
         champion.enemyVictory();
     }
 
     public void championAttack() {
         System.out.print("Champion attack: ");
         monster.setHp(monster.getHp() - champion.championDamage());
-        System.out.println("Monster have now " + monster.getHp() + " hit points");
-        monster.enemyVictory();
-        if (!monster.isDefeat()) {
+        System.out.println("The " + monster.getType().charType + " have now " + monster.getHp() + " hit points");
+        if (monster.getType().equals(CharacterTypes.GOBLIN_KING) && monster.getHp() <= 0) {
+            System.out.println("The goblin king is dead! You win!");
+            goblinKingIsDeadTest = true; //toTest
```

Comment:
You can test goblin king defeat without this.

Reply:
Tests reworked. This boolen removed